robotoff/insights/annotate.py

```diff
@@ -107,11 +108,12 @@ def _annotate(
         if automatic:
             insight.automatic_processing = True
 
-        insight.save()
-
         if annotation == 1 and update:
             return self.process_annotation(insight, data=data, auth=auth)
 
+        insight.annotated_result = 1
+        insight.save
+
         return SAVED_ANNOTATION_RESULT
```

Comment:
several comments:

1. you have to call the method save ;-) --> `insight.save()`
2. by moving this line under the `if` where there is a return, you are changing code logic. So you have to move it back on top
3. why use "1", when what we want to save is the returned value ! here, SAVE_ANNOTATION_RESULT

I would write:

```suggestion
        insight.save()
        if annotation == 1 and update:
            annotation_result = self.process_annotation(insight, data=data, auth=auth)
        else:
            annotation_result = SAVED_ANNOTATION_RESULT
        insight.annotated_result = 1
        insight.save()  # second call to save annotated_result
        return annotation_result
```

Reply:
Ok, reading what you wrote below I understand you changes. 

That said, it's really better to keep things centralized and avoid spread logic for a specific detail all around the code. So I think this proposed pattern is better (even if it requires two "save" call)